src/entity.ts

```diff
@@ -152,9 +157,36 @@ export class Entity {
 		}
 	}
 
-	set(properties: ObjectLookup<any>): void;
-	set(property: string, value: any): void;
-	set(property: any, value?: any): void {
+	private static createOrUpdate(type: Type, state: any, context?: InitializationContext) {
+		const id = getIdFromState(type, state);
+		const isNew = !id;
+		if (!context)
+			context = new InitializationContext(isNew);
+
+		// We need to pause processing of callbacks to prevent publishing entity events while still processing
+		// the state graph
+		const instance = context.execute(() => {
+			let instance = id && type.get(id);
+			if (instance) {
+				// Assign state to the existing object
+				instance.updateWithContext(context, state);
+			}
+			else {
+				// Cast the jstype to any so we can call the internal constructor signature that takes a context
+				// We don't want to put the context on the public constructor interface
+				const Ctor = type.jstype as any;
+				// Construct an instance using the known id if it is present
+				instance = (id ? new Ctor(id, state, context) : new Ctor(state, context)) as Entity;
+			}
+			return instance;
+		});
+
+		return instance;
+	}
+
```

Comment:
This function was pretty much a straight copy from `Type$createOrUpdate`.

Reply:
Looks like the only difference was getting rid of the `contextOrResolver` logic and have it just be a context. Which I think was no longer needed since you made value resolvers a model concept.